pilot/pkg/networking/v1alpha3/listener.go

```diff
@@ -396,6 +392,33 @@ func buildMgmtPortListeners(managementPorts model.PortList, managementIP string)
 	return listeners
 }
 
+// buildMixerFilter builds a filter with a v1 mixer config encapsulated as JSON in a proto.Struct for v2 consumption.
+func buildMixerFilter(env model.Environment, node model.Proxy,
+	proxyInstances []*model.ServiceInstance, protocol model.Protocol, outbound bool /*instance *model.ServiceInstance*/) listener.Filter {
+	mesh := env.Mesh
+	config := env.IstioConfigStore
+	if mesh.MixerCheckServer == "" && mesh.MixerReportServer == "" {
+		return listener.Filter{}
+	}
+
+	mixerFilter := listener.Filter{}
+	switch protocol {
+	case model.ProtocolHTTP, model.ProtocolHTTP2, model.ProtocolGRPC:
+		config := v1.BuildHTTPMixerFilterConfig(mesh, node, proxyInstances, outbound, config)
```

Comment:
Lets please not cross into v1. I would much rather copy that code here, the way I did for TCP proxy routes. I would like to be able to delete all proxy/v1 and still see the mesh functioning.